docs/transformations.md

```diff
@@ -0,0 +1,91 @@
+# Transformations {#transformations}
+<!--
+Copyright (C) The DDC development team, see COPYRIGHT.md file
+
+SPDX-License-Identifier: MIT
+-->
+
+On the top of DDC core - which provides the basic tools for the needs of performance-portable physical simulations
+(describing meshes with *DiscreteDomain* and representing physical fields with *Chunk*) - are build computation kernels
+to perform transformations. At the moment, the two available transformations are Discrete Fourier Transform and Spline methods.
+
+Those methods are performance portable (supported with CPU Serial, CPU OpenMP, CUDA and HIP backends). They are used
+in *examples/heat_equation_spectral.cpp* (for DFT) and *examples/characteristics_advection.cpp* (for splines).
+A particular use-case of splines is benchmarked in *benchmarks/splines.cpp*.
+
+## General presentation of transformations
+
+### Computation kernels dedicated to transformations in DDC
+
+A transformation is a change of basis, going from a function space to another. Because the available memory is finite,
+the basis are necessarily finite (which means it contains a finite set of basis functions, but those basis functions
+may be continuous).
+
+The use-cases of transformations for simulation or signal processing needs are plethoric: interpolation, spectral methods,
+finite element methods, filtering (compression), signal analysis (post-process)...
+
+Every basis have their own specificities (ie. orthogonality, n-derivability, etc...) but as they belong to the same frame
+of set of functions which form a basis, they have in common a formalism and a terminology. However, this is currently
+not very manifest when looking at the API in the DDC implementations (DFT and Spline API are very different).
+It must be explained:
+
+- Fourier requires the periodicity of the represented function, thus boundary conditions does not need to be provided.
+- Fourier basis functions are indexed with the wave vector k, whose possible values form a set of coordinates which generates
+the Fourier space. Mesh of "real" space and mesh of Fourier space are in bijection one-to-the-other.
+The situation is more complicated for Splines.
+- Fourier and Splines basis use-cases are quite different. Thus - as shown below - the so-called "evaluator" is not
+implemented for Fourier (while inverse DFT is) and inverse Spline transform is not implemented (while Spline evaluation is).
+- FFT backends support 1D to 3D Fourier transforms. General n-dimensional Spline transform would be a mess to implement for all supported boundary conditions so we limit to the 2D case.
+- DFT is not currently supported in batched configuration (but it may be the subject of futur development). Batched splines are supported.
```

Comment:
```suggestion
- DFT is not currently supported in batched configuration (but it may be the subject of future development). Batched splines are supported.
```